x-pack/agent/docs/agent_configuration_example.yml

```diff
@@ -0,0 +1,518 @@
+            metricset: billing
+            dataset: aws.billing
+            period: 12
+          - id?: {id}
+            metricset: s3_daily_storage
+            dataset: aws.s3_daily_storage
+            period: 24
+          - id?: {id}
+            metricset: s3_request
+            dataset: aws.s3_request
+            period: 24
+
+
+  #################################################################################################
+  # - Kubernetes
+  - id?: my-kubernetes
+    title: Collect Kubernetes
+    package:
+      name: epm/kubernetes
+      version: 1.7.0
+    inputs:
+      - type: metrics/kubernetes
+        hosts: ["localhost:10250"]
+        bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
+        ssl.certificate_authorities:
+          - /var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt
+        # username: "user"
+        # password: "secret"
+        # If kube_config is not set, KUBECONFIG environment variable will be checked
+        # and if not present it will fall back to InCluster
+        # kube_config: ~/.kube/config
+        streams:
+          - id?: {id}
+            metricset: node
+            dataset: kubernetes.node
+            period: 10s
+          - id?: {id}
+            metricset: system
+            dataset: kubernetes.system
+            period: 10s
+          - id?: {id}
+            metricset: pod
+            dataset: kubernetes.pod
+            period: 10s
+          - id?: {id}
+            metricset: container
+            dataset: kubernetes.container
+            period: 10s
+          - id?: {id}
+            metricset: container
+            dataset: kubernetes.container
+            period: 10s
+          - id?: {id}
+            metricset: event
+            dataset: kubernetes.event
+            period: 10s
+      - type: metrics/kubernetes
+        hosts: ["kube-state-metrics:8080"]
+        streams:
+          - id?: {id}
+            metricset: state_node
+            dataset: kubernetes.state_node
+            period: 10s
+          - id?: {id}
+            metricset: state_deployment
+            dataset: kubernetes.state_deployment
+            period: 10s
+          - id?: {id}
+            metricset: state_replicaset
+            dataset: kubernetes.state_replicaset
+            period: 10s
+          - id?: {id}
+            metricset: state_statefulset
+            dataset: kubernetes.state_statefulset
+            period: 10s
+          - id?: {id}
+            metricset: state_pod
+            dataset: kubernetes.state_pod
+            period: 10s
+          - id?: {id}
+            metricset: state_container
+            dataset: kubernetes.state_container
+            period: 10s
+          - id?: {id}
+            metricset: state_container
+            dataset: kubernetes.state_container
+            period: 10s
+          - id?: {id}
+            metricset: state_cronjob
+            dataset: kubernetes.state_cronjob
```

Comment:
To make it a bit more magical, this is `kubernetes.cronjob`. Same for the several above ;-)